src/unpoly/classes/request/xhr_renderer.js

```diff
@@ -10,7 +10,7 @@ up.Request.XHRRenderer = class XHRRenderer {
   }
 
   buildAndSend(handlers) {
-    this.xhr = new XMLHttpRequest()
+    this.xhr = this.request.xhr
```

Comment:
`this.xhr` can just be a local variable now, which compresses better.

Also we no longer need to return the XHR object at the end of the method.

Reply:
The xhr is now local, which also required to pass xhr to the addHeader method.